rfcs/0005-artifact-revision-and-digest/README.md

```diff
@@ -0,0 +1,367 @@
+derived from `GitRepository` which uses `/` as a separator between the named
+pointer (a Git branch or tag) and a specific (SHA-1, or theoretical SHA-256)
+revision.
+
+Since the introduction of `OCIRepository` and with the recent changes around
+`HelmChart` objects to allow the consumption of charts from OCI registries,
+this could be seen as outdated or confusing due to the format differing from
+the canonical format used by OCI, which is `<name>@<algo>:<checksum>` (the
+part after `@` formally known as a ["digest"][digest-spec]) to refer to a
+specific version of an OCI manifest.
+
+While also taking note that Git does not have an official canonical format for
+e.g. branch references at a specific commit, and `/` has less of a symbolic
+meaning than `@`, which could be interpreted as "`<branch>` _at_
+`<commit SHA>`".
+
+In addition, with the introduction of algorithm prefixes for an `Artifact`'s
+checksum, it would be possible to add support and allow user configuration of
+other algorithms than SHA-256. For example SHA-384 and SHA-512, or the more
+performant (parallelizable) [BLAKE3][].
+
+Besides this, it would make it easier to implement a client that can verify the
+checksum without having to resort to an assumed format or guessing
+method based on the length of it, and allows for a more robust solution in
+which it can continue to calculate against the algorithm of a previous
+configuration.
+
+The inclusion of the `Artifact`'s algorithm prefix has been proposed before in
+[source-controller#855](https://github.com/fluxcd/source-controller/issues/855),
+with supportive response from Core Maintainers.
+
+### Goals
+
+- Establish a canonical format to refer to an `Artifact`'s `Revision` field
+  which consists of a named pointer and a specific checksum reference.
+- Allow easier verification of the `Artifact`'s checksum by including an
+  alias for the algorithm.
+- Deprecate the `Artifact`'s `Checksum` field in favor of the `Digest` field.
+- Allow configuration of the algorithm used to calculate the checksum of an
+  `Artifact`.
+- Allow configuration of algorithms other than SHA-256 to calculate the
+  `Digest` of an `Artifact`.
+- Allow compatibility with SemVer name references which might contain an `@`
+  symbol already (e.g. `package@v1.0.0@sha256:...`, as opposed to OCI's
+  `name:v1.0.0@sha256:...`).
+
+### Non-Goals
+
+- Define a canonical format for an `Artifact`'s `Revision` field which contains
+  a named pointer and a different reference than a checksum.
+
+## Proposal
+
+### Establish an Artifact Revision format
+
+Change the format of the `Revision` field of the `source.toolkit.fluxcd.io`
+Group's `Artifact` type across all `Source` kinds to contain an `@` separator
+opposed to `/`, and include the algorithm alias as a prefix to the checksum
+(creating a "digest").
+
+```text
+[ <named pointer> ] [ [ "@" ] <algo> ":" <checksum> ]
+```
+
+Where `<named pointer>` is the name of e.g. a Git branch or OCI repository
```

Comment:
Having the named pointer either be the branch name or OCI repository is a bit weird in my head. These two values represent different things. Its like including the repository URL in the named pointer instead of the Git branch.

Reply:
The scope does not explicitly prohibit having a format like e.g. `image-name:tag@sha256:62d3f07430651f515089c402fd006fc0c63382a7fc1087adda3becbb6b4481a3`.

Reply:
Ok sounds good.